### PR TITLE
[Fil-480] Add validation to add storage providers input

### PR DESCRIPTION
**Pull request by Filip-L**

### CODE REVIEW

File: src/components/cards/dialogs/allowedSps/AllowedSps.tsx [hunk 1]
```diff
@@ -83,14 +85,40 @@ export const AllowedSPs: React.FC<ComponentProps> = ({
...
+
   const handleInputChange = (index: number, value: string): void => {
     const newData = [...data]
+    const updatedErrors = [...errors]
     newData[index] = value
+    value = value.trim()
+    clearTimeout(debounceTimer)
+    const timer = window.setTimeout(() => {
+      value = value.trim()
+      console.log(value)
```
Comment (kacperzuk-neti) [inline src/components/cards/dialogs/allowedSps/AllowedSps.tsx:101]: Cleanup?

File: src/components/cards/dialogs/allowedSps/AllowedSps.tsx [hunk 2]
```diff
@@ -83,14 +85,40 @@ export const AllowedSPs: React.FC<ComponentProps> = ({
...
+      if (!value) {
+        updatedErrors[index] = 'Value cannot be empty'
+        setErrors([...updatedErrors])
+        return
+      } else if (!isValidInput(value)) {
+        updatedErrors[index] = 'Invalid storage provider id'
+        setErrors([...updatedErrors])
```
Comment (kacperzuk-neti) [inline src/components/cards/dialogs/allowedSps/AllowedSps.tsx:108]: @tplocic20 what do you think about DYI form validation and debouncing? Is this the best way given the state of the app or is there some good lightweight lib we could include?

**Review by kacperzuk-neti — APPROVED** [review]
Get approval from @tplocic20 before merging please.

File: src/components/cards/dialogs/allowedSps/AllowedSps.tsx (hunk 2 above)
Comment (tplocic20) [reply src/components/cards/dialogs/allowedSps/AllowedSps.tsx:107]: Ideally we would use something like hookForm - but it's too much of a refactor I'm afraid

**Review by tplocic20 — APPROVED** [review]

**Pull request merged**